From c5bcd1a68bc74453c698affea0b62c545dd782c6 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Sat, 2 Mar 2024 14:32:07 +0100
Subject: [PATCH 01/14] phy: rockchip: usbdp: Change init/exit callbacks to
 power_on/off

DWC3 driver calls power_on/off during host/device mode changes. Init/
exit are only called once per DWC3 device probe/remove lifecycle.

These callbacks need to be called during mode change to actually re-
configure the PHY.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 drivers/phy/rockchip/phy-rockchip-usbdp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/phy/rockchip/phy-rockchip-usbdp.c b/drivers/phy/rockchip/phy-rockchip-usbdp.c
index fba35510d88c..73090bbb19cb 100644
--- a/drivers/phy/rockchip/phy-rockchip-usbdp.c
+++ b/drivers/phy/rockchip/phy-rockchip-usbdp.c
@@ -1336,8 +1336,8 @@ static int rk_udphy_usb3_phy_exit(struct phy *phy)
 }
 
 static const struct phy_ops rk_udphy_usb3_phy_ops = {
-	.init		= rk_udphy_usb3_phy_init,
-	.exit		= rk_udphy_usb3_phy_exit,
+	.power_on	= rk_udphy_usb3_phy_init,
+	.power_off	= rk_udphy_usb3_phy_exit,
 	.owner		= THIS_MODULE,
 };
 
-- 
2.43.0

